esp/platform/txsummary.cpp

```diff
@@ -65,57 +356,112 @@ bool CTxSummary::contains(const char* key) const
     return __contains(key);
 }
 
-bool CTxSummary::append(const char* key, const char* value, const LogLevel logLevel)
+bool CTxSummary::appendSerialized(const char* key, const char* value, const LogLevel logLevel, const unsigned int group, bool jsonQuoted, const char* suffix)
 {
-    VALIDATE_KEY(key);
+    if(!validate(key))
+        return false;
 
     CriticalBlock block(m_sync);
 
     if (__contains(key))
         return false;
 
-    m_entries.push_back({key, value, logLevel});
+    EntryValue entry;
+    entry.setown(new TxEntryStr(key, value, logLevel, group, jsonQuoted, suffix));
+    m_entries.push_back(entry);
+
     return true;
 }
 
-bool CTxSummary::set(const char* key, const char* value, const LogLevel logLevel)
+// Searches for the entry named 'key' and replaces it with the new entry 'value'
+// If not found, then it adds the new entry 'value' to the end of the list
+bool CTxSummary::setEntry(const char* key, TxEntryBase* value)
 {
-    VALIDATE_KEY(key);
-
-    CriticalBlock block(m_sync);
-    Entries::iterator it = find_if(m_entries.begin(), m_entries.end(), MATCH_KEY);
+    if(!validate(key))
+         return false;
 
-    if (it != m_entries.end())
+    bool found = false;
+    EntryValue entry;
+    entry.setown(value);
+    for(auto entryItr = m_entries.begin(); entryItr != m_entries.end(); entryItr++)
     {
-        it->value.set(value);
-        it->logLevel = logLevel;
+        if(strieq(key, entryItr->get()->fullname.str()))
+        {
+            found = true;
+            auto finger = m_entries.erase(entryItr);
```

Comment:
Since I can't reply to your response, I'm going to add a new comment.

I think it is inconsistent for a timer to be replaceable by any other value because of a name collision that didn't previously exist and a presumption that any new value is more important than a potentially active timer, while at the same time suggesting that a new timer cannot be as important as any other new value.

While setEntry is private, it is called from a public method. That public method is not exposed through IEspContext, but the summary's usage is not limited to IEspContext.

Reply:
I will address this in a subsequent PR. The resolutions I tried either didn't work as expected or had cascading effects that would have resulted in a bigger change than manageable now.